cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1860,32 +1862,25 @@ instantiateInstallPlan plan =
 
 
 --TODO: this needs to report some user target/config errors
-elaboratePackageTargets :: ElaboratedConfiguredPackage -> [PackageTarget]
-                        -> ([ComponentTarget], Maybe ComponentTarget, Bool)
-elaboratePackageTargets ElaboratedConfiguredPackage{..} targets =
-    let buildTargets  = nubComponentTargets
-                      . concatMap elaborateBuildTarget
-                      $ targets
-        --TODO: instead of listToMaybe we should be reporting an error here
-        replTargets   = listToMaybe
-                      . nubComponentTargets
-                      . concatMap elaborateReplTarget
-                      $ targets
-        buildHaddocks = HaddockDefaultComponents `elem` targets
-
-     in (buildTargets, replTargets, buildHaddocks)
+elaboratePackageTargets :: ElaboratedInstallPlan
+                        -> Map UnitId [PackageTarget]
+                        -> Map UnitId [ComponentTarget]
+elaboratePackageTargets elaboratedPlan =
+    Map.mapWithKey $ \pkgid targets ->
+      let Just (InstallPlan.Configured pkg) =
```

Comment:
Note that this is only an intermediate step during the refactoring. This function gets completely removed in subsequent patches. It's replaced by the combination of `availableTargets` and `resolveTargets` which are much more careful about error checking.